scripts/quests/sandoria/Tigers_Teeth.lua

```diff
@@ -0,0 +1,99 @@
+-----------------------------------
+-- Tigers Teeth
+-----------------------------------
+-- LogID: 0 QuestID: 23
+-- Taukila : !pos -140 -6 -8 230
+-----------------------------------
+
+local quest = Quest:new(xi.questLog.SANDORIA, xi.quest.id.sandoria.TIGER_S_TEETH)
+
+quest.reward =
+{
+    title = xi.title.FANG_FINDER,
+    fame = 25,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_AVAILABLE and
+                player:getFameLevel(xi.fameArea.SANDORIA) >= 3
+        end,
+
+        [xi.zone.SOUTHERN_SAN_DORIA] =
+        {
+            ['Taumila'] = quest:progressEvent(574),
+
+            onEventFinish =
+            {
+                [574] = function(player, csid, option, npc)
+                    if option == 0 then
+                        quest:begin(player)
+                    end
+                end,
+            },
+        },
+    },
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.SOUTHERN_SAN_DORIA] =
+        {
+            ['Taumila'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, { { xi.item.BLACK_TIGER_FANG, 3 } }) then
+                        return quest:progressEvent(572)
+                    elseif npcUtil.tradeHas(trade, xi.item.BLACK_TIGER_FANG) then
+                        return quest:event(573)
+                    end
+                end,
+
+                onTrigger = quest:event(575):replaceDefault(),
+            },
+
+            onEventFinish =
+            {
+                [572] = function(player, csid, option, npc)
+                    quest:complete(player)
+                end,
```

Comment:
Shouldnt this complete the trade aswell?

Reply:
explained up top it compltes here and still runs the event finish 572 in the quest_completed section. just a cool trickle down effect and why i have the fame for completes at 25 and you get the 5 fame in the completed section to addup to 30, the completed trade hits there as well

Reply:
Could you add a comment in here stating that both onEventFinish, both in different sections, execute?

Its a bit counter-intuitive, tbh. Not saying its wrong, but it has certanly taken me by surprise

Reply:
i changed it to local var checks and not use the trickle down effect separating the sections